src/test/java/io/stargate/sgv2/jsonapi/api/v1/IndexingConfigIntegrationTest.java

```diff
@@ -0,0 +1,689 @@
+          .then()
+          .statusCode(200)
+          .body("status", is(nullValue()))
+          .body("data", is(nullValue()))
+          .body("errors[0].message", endsWith("The filter path ('address.city') is not indexed"))
+          .body("errors[0].errorCode", is("UNINDEXED_FILTER_PATH"))
+          .body("errors[0].exceptionClass", is("JsonApiException"));
+    }
+
+    @Test
+    public void filterIdInDenyAllWithEqAndIn() {
+      // deny all use "*"
+      String filterId1 =
+          """
+                  {
+                    "find": {
+                      "filter": {"_id": "1"}
+                    }
+                  }
+                    """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(filterId1)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, denyAllIndexingCollection)
+          .then()
+          .statusCode(200)
+          .body("status", is(nullValue()))
+          .body("errors", is(nullValue()))
+          .body("data.documents", hasSize(1));
+
+      String filterId2 =
+          """
+                {
+                    "find": {
+                        "filter": {
+                        }
+                    }
+                }
+                    """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(filterId2)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, denyAllIndexingCollection)
+          .then()
+          .statusCode(200)
+          .body("status", is(nullValue()))
+          .body("errors", is(nullValue()))
+          .body("data.documents", hasSize(1));
+    }
+
+    @Test
+    public void filterIdInDenyAllWithoutEqAndIn() {
+      // deny all use "*"
+      String filterId3 =
+          """
+                {
+                    "find": {
+                        "filter": {
+                            "_id": {
+                                "$nin": [
+                                    "1",
+                                    "2"
+                                ]
+                            }
+                        }
+                    }
+                }
+                    """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(filterId3)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, denyAllIndexingCollection)
+          .then()
+          .statusCode(200)
+          .body("status", is(nullValue()))
+          .body("data", is(nullValue()))
+          .body(
+              "errors[0].message",
+              endsWith(
+                  "The filter path ('_id') is not indexed, you can only use $eq or $in as the operator"))
```

Comment:
I'm a bit confused by this error message. The test on line 351 tests that you cannot query on a string field if the field isn't indexed. But this error message indicates that you should be able to query on a non-indexed field as long as you're using `$eq`. Can you please explain the difference?

Reply:
If a field name is in the deny list or not in the allow list (implicitly denied), it cannot be used in `filter`. The exception is `_id`. If `_id` is denied, it can only be used with `$eq` or `$in` in `filter`. The reason is:
```
Denying _id
The Document Identity field, _id, can be denied from the index. This can be done by either listing it explicitly in the deny list, or using the wildcard pattern in the deny list. 

This is possible because the _id field has two processing paths in the API: 
1. It is the CQL Table primary key, stored as a tuple of (data_type, value). 
2. It is indexed through super shredding using the appropriate data type. 

Read queries decide which representation to use based on the filter: 
1. When _id is the only term in a filter, and the operation is either $eq or $in the CQL Table primary key is used as the fast path lookup to get the row. 
2. When _id is used with other terms, or other operations, the super shredding representation may be used . 

As a result, when _id is denied from the index it is still possible to filter documents by _id using either the $eq or $in operations.  Further, it is legal for a collection to deny all fields from the index using the wildcard match to create a collection that only supports reading by the document id. This would use significantly less disk space, cost less for the end user, and still support projections and updates on the JSON document.
```

Reply:
Thanks for clarifying, that makes sense